[wip] feat(text randomizer): implement text randomizer and toggle between wheel

imballinst opened the pull request:
Closes https://github.com/ans-4175/random-gofood/issues/10. More PR description TBD.

imballinst commented:
@ans-4175 ada ide ngga untuk ngedengerin event change di tabsnya? Tidak terlihat ada propsnya di docs ini: https://github.com/rough-stuff/wired-elements/blob/master/docs/wired-tabs.md#wiredtabs-properties

Terus saya entah kenapa dapet eror ini:

![image](https://user-images.githubusercontent.com/7077157/142438679-b0d762df-c8af-4e35-b157-8b5365409d82.png)

padahal di Codesandbox seperti aman: https://codesandbox.io/s/react-wrapper-for-wired-elements-forked-fef11?file=/src/index.js

ans-4175 commented:
> hf.fillPolygon

Ini kemarin aku kena isu ini di Roulette, bukan di wired element kalau tak salah.

ans-4175 commented:
> listen event tab

https://github.com/rough-stuff/wired-elements/blob/master/src/wired-tabs.ts
Disini dia ga ada `fireEvent` euy, emang dia ga ngebuang sesuatu, bagaimana kalau itu dipasang `ref` dan baca perubahan nilai property selected-nya di `wiredTabs`

ans-4175 commented on src/App.js:
```diff
@@ -42,19 +45,27 @@ import DetailMerchant from './components/DetailMerchant';
+  // `undefined` states mean not picked yet.
+  // When it is defined, then it means these 2 states have been picked/fetched.
+  /** @type [Merchant | undefined, Function] */
+  const [pickedMerchant, setPickedMerchant] = useState(undefined);
```
nah ini coba ganti jangan undefined, tapi empty object lagi aja?

imballinst replied:
bisa sih, berarti state `fetched` balikin juga yak?

ans-4175 replied:
hmm, tapi bener kamu sih secara semantic.
Tadi aku berpikir yang salah render itu apa.
kalau roulette, ga akan kena disini ya kan?

imballinst replied:
iyak, roulette harusnya ngga kena, soalnya dia nerimanya `prizeNumber` bukan `pickedMerchant`

ans-4175 replied:
okay kl gitu stay `undefined` aja ji, lebih semantic wise

imballinst replied:
siap :+1:

imballinst commented:
> > listen event tab
> 
> https://github.com/rough-stuff/wired-elements/blob/master/src/wired-tabs.ts Disini dia ga ada `fireEvent` euy, emang dia ga ngebuang sesuatu, bagaimana kalau itu dipasang `ref` dan baca perubahan nilai property selected-nya di `wiredTabs`

OK, ini bisa dicoba

ans-4175 commented on src/App.js:
```diff
@@ -169,47 +188,58 @@ function App() {
             <section>
-              {wheelData && (
-                <>
+              <WiredTabs selected="wheel">
```
nah ini coba dipasang property selected-nya kalau berubah gitu

imballinst replied:
o ternyata dia kalau hasil build dengan `NODE_ENV=production` ngga eror ya? Cuma di lokal aja erornya... menarik

ans-4175 replied:
error yang selected?
Aku belum nyambung

imballinst replied:
iyak, ini kalau di lokal aku, dia bakal ngecrash

tapi kalau di https://random-gofood-git-fork-imballinst-10-tabs-randomizer-ans4175.vercel.app/, dia nggak nge-crash, kenapa yak? padahal kodenya sama

ans-4175 replied:
kalau aku dulu, itu masalah si pas yarn start-nya, kl dijalanin ulang lagi tetap masalah?

imballinst replied:
iyak, dijalanin ulang tetep masalah. setelah saya liat lagi ternyata dia kalau di prod tetep ada erornya yang sama, tapi ngga bikin crash:

![image](https://user-images.githubusercontent.com/7077157/142448090-5c68201d-6288-466c-9b31-409f9377902d.png)

ans-4175 commented on src/App.js:
```diff
@@ -169,47 +188,58 @@ function App() {
+                <p>Waiting to spin...</p>
+              ) : (
+                <div>
+                  <WiredButton elevation={2} onClick={handleSpinClick}>
```
jadi ini idenya saat tekan SPIN, semua randomized?

imballinst replied:
"semua randomized" ini boleh diperjelas, Kak Ans? Bagian ini nggak saya ubah sih (cuma saya keluarin aja dari `wheelData`, soalnya `wheelData` pasti bakal `true` karena dia default valuenya empty array).

Tapi nanti mau saya ubah, kayak ketika tabnya teks, maka dia tulisannya "Randomize", dan ketika tabnya wheel, dia "Spin".

ans-4175 replied:
maksudnya 2 mode, spin atau textbllinker itu berjalan. ga kan?

imballinst replied:
oooh, hmmm. actually sekarang 2 2nya jalan sih, soalnya mereka pakai state yang sama.

mungkin nanti saya flag supaya tab yang nggak aktif, komponen randomizernya unmount (jadi randomizer yang jalan hanya yang tabnya aktif).

ans-4175 replied:
okay gpp sih, render-nya juga no effort kan site-nya..wkwk

imballinst commented:
@ans-4175 saya isi `App.js` cuma ini masih ngecrash:

```
import { WiredTabs, WiredTab } from 'wired-elements-react';

import './App.css';

function App() {
  return (
    <WiredTabs>
      <WiredTab name="Wheel">test</WiredTab>
      <WiredTab name="Text">test2</WiredTab>
    </WiredTabs>
  );
}

export default App;
```

saya coba nuke terus install ulang `node_modules` deh

imballinst commented:
wah, kurang paham, saya download file Sandbox https://codesandbox.io/s/react-wrapper-for-wired-elements-forked-fef11?file=/src/index.js, install, lalu jalanin, tetep crash :sweat_smile: mungkin harus dibawa tidur dulu baru bisa jalan

@ans-4175 kalau sempat, boleh tolong coba jalanin PR ini di lokal nggak? atau download lalu jalanin Sandbox :point_up: di lokal? takutnya ada masalah dependency yang aneh-aneh kayak https://github.com/apollographql/apollo-link-state/issues/187

imballinst commented:
Oh, kayaknya ini penyebabnya: https://github.com/rough-stuff/rough/commit/24fd61d3be2dd411f4153b2885ed3d912a3b56fd#diff-054c66f0fb4838b0ff226ca4dea0ebcc007364ff1baf314d788a8015cbac108a

Dia ngeresolve packagenya ke package versi yang terbaru, yang mana `fillPolygon` udah berubah jadi `fillPolygons`. Mungkin kita bisa kasih `resolutions` di `package.json` sehingga dia selalu ngarah ke versi sebelumnya

EDIT: hore bisa

![image](https://user-images.githubusercontent.com/7077157/142452989-52000ff0-2117-4076-b2b4-1d82595e24e7.png)

ans-4175 commented:
<img width="950" alt="image" src="https://user-images.githubusercontent.com/3961872/142453855-16e7cc00-9817-459b-a439-430fee424d01.png">
Iyah aku coba di codespaces juga error

ans-4175 commented:
> Dia ngeresolve packagenya ke package versi yang terbaru

Oh oke, fixed?

imballinst commented:
> > Dia ngeresolve packagenya ke package versi yang terbaru
> 
> Oh oke, fixed?

@ans-4175 iyak, fixed. Saya submit bugnya di https://github.com/rough-stuff/wired-elements/issues/179. Nanti saya follow-up di sini pakai Yarn module resolutions ke `"roughjs": "4.4"` untuk sementara waktu.

imballinst commented:
@ans-4175 saya boleh pake https://www.npmjs.com/package/patch-package aja nggak? Kayaknya lebih straightforward daripada kalau harus nambahin ref lalu listen childrennya... mungkin kalau mau nanti patchnya bisa dikontribusiin juga ke `wired-elements` dan `wired-elements-react`

## wired-elements-react

```
diff --git a/node_modules/wired-elements-react/lib/WiredTabs.d.ts b/node_modules/wired-elements-react/lib/WiredTabs.d.ts
index cbafd3d..c37934b 100644
--- a/node_modules/wired-elements-react/lib/WiredTabs.d.ts
+++ b/node_modules/wired-elements-react/lib/WiredTabs.d.ts
@@ -1,5 +1,7 @@
 import * as React from 'react';
 import { WiredTabs as CE } from 'wired-elements/lib/wired-tabs.js';
-export declare const WiredTabs: React.ForwardRefExoticComponent<Partial<CE> & {} & {
+export declare const WiredTabs: React.ForwardRefExoticComponent<Partial<CE> & {
+    onselected?: ((e: Event) => unknown) | undefined;
+} & {
     children?: React.ReactNode;
 } & React.RefAttributes<unknown>>;
diff --git a/node_modules/wired-elements-react/lib/WiredTabs.js b/node_modules/wired-elements-react/lib/WiredTabs.js
index 6c93459..461610a 100644
--- a/node_modules/wired-elements-react/lib/WiredTabs.js
+++ b/node_modules/wired-elements-react/lib/WiredTabs.js
@@ -1,4 +1,6 @@
 import * as React from 'react';
 import { createComponent } from '@lit-labs/react';
 import { WiredTabs as CE } from 'wired-elements/lib/wired-tabs.js';
-export const WiredTabs = createComponent(React, 'wired-tabs', CE);
+export const WiredTabs = createComponent(React, 'wired-tabs', CE, {
+  onselected: 'selected'
+});
```

## wired-elements

```
diff --git a/node_modules/wired-elements/lib/wired-lib.js b/node_modules/wired-elements/lib/wired-lib.js
index 5b1d153..9d3ec39 100644
--- a/node_modules/wired-elements/lib/wired-lib.js
+++ b/node_modules/wired-elements/lib/wired-lib.js
@@ -90,7 +90,7 @@ export function ellipse(parent, x, y, width, height, seed) {
 }
 export function hachureFill(points, seed) {
     const hf = new ZigZagFiller(fillHelper);
-    const ops = hf.fillPolygon(points, options(seed));
+    const ops = hf.fillPolygons([points], options(seed));
     return createPathNode(ops, null);
 }
 export function hachureEllipseFill(cx, cy, width, height, seed) {
diff --git a/node_modules/wired-elements/lib/wired-tabs.js b/node_modules/wired-elements/lib/wired-tabs.js
index f9dfb28..58c8def 100644
--- a/node_modules/wired-elements/lib/wired-tabs.js
+++ b/node_modules/wired-elements/lib/wired-tabs.js
@@ -7,7 +7,7 @@ var __decorate = (this && this.__decorate) || function (decorators, target, key,
 var __metadata = (this && this.__metadata) || function (k, v) {
     if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
 };
-import { BaseCSS } from './wired-base';
+import { fireEvent, BaseCSS } from './wired-base';
 import { css, html, LitElement } from 'lit';
 import { customElement, property, query } from 'lit/decorators.js';
 let WiredTabs = class WiredTabs extends LitElement {
@@ -149,6 +149,7 @@ let WiredTabs = class WiredTabs extends LitElement {
                 index--;
             }
             this.selected = list[index].name || '';
+            this.fireSelected();
         }
     }
     selectNext() {
@@ -171,8 +172,12 @@ let WiredTabs = class WiredTabs extends LitElement {
                 index++;
             }
             this.selected = list[index].name || '';
+            this.fireSelected();
         }
     }
+    fireSelected() {
+        fireEvent(this, 'selected', { selected: this.selected });
+    }
 };
 __decorate([
     property({ type: String }),
```

ans-4175 commented:
Aku belum pernah pakai
Itu gimana ya?
Possible aja ya

On Fri, 19 Nov 2021 at 19.59 Try Ajitiono ***@***.***> wrote:

> @ans-4175 <https://github.com/ans-4175> saya boleh pake
> https://www.npmjs.com/package/patch-package aja nggak? Kayaknya lebih
> straightforward daripada kalau harus nambahin ref lalu listen
> childrennya... mungkin kalau mau nanti patchnya bisa dikontribusiin juga ke
> wired-elements dan wired-elements-react
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/ans-4175/random-gofood/pull/14#issuecomment-974050005>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA6HIEFD3NZ3RQWK42U2LYLUMZCZRANCNFSM5IJ2CPUQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
-- 
Ahmad Anshorimuslim Syuhada
Data Engineer - Web Developer - Linux Enthusiast
+6285224752182 - @ans4175
"trust me I'm [already] an engineer"

imballinst commented:
@ans-4175 kira-kira bagusan mana? yang pertama atau yang kedua?

![image](https://user-images.githubusercontent.com/7077157/142627661-faaf6b68-06d0-4d49-aed0-9da673e2e679.png)

![image](https://user-images.githubusercontent.com/7077157/142627672-1e5d2fe4-0387-402d-a4cf-05dcc06a1d59.png)

yang pertama yang bawaan, yang kedua yang saya tweak. kok saya ngerasanya ada "border" di sekitar wheel itu kurang cocok ya? soalnya dia udah di atas elemen wired anyway... apakah elemen wired di atas elemen wired itu design language yang masuk?

imballinst commented:
> Aku belum pernah pakai
> Itu gimana ya?
> Possible aja ya

@ans-4175 jadi kita nge-update `node_modules`, nanti si `patch-package` ini bakal nge-output diff dari file-file yang kita update. setiap kali kita pakai command `yarn`, dia bakal re-apply diff-diff tersebut, jadi hasilnya konsisten

ans-4175 commented:
Kita ngunci di paketnya gitu ya terus bikin diff
Boleh deh cobain
Aku ga terlalu kebayang ji

Anyway, iyah tanpa border lebih masuk pas kayaknya
Itu pakai wired card buatnya? Kok aneh ya

On Fri, 19 Nov 2021 at 20.10 Try Ajitiono ***@***.***> wrote:

> Aku belum pernah pakai
> Itu gimana ya?
> Possible aja ya
>
> @ans-4175 <https://github.com/ans-4175> jadi kita nge-update node_modules,
> nanti si patch-package ini bakal nge-output diff dari file-file yang kita
> update. setiap kali kita pakai command yarn, dia bakal re-apply diff-diff
> tersebut, jadi hasilnya konsisten
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/ans-4175/random-gofood/pull/14#issuecomment-974058638>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA6HIEE3X4VH5P5CGIZRV3DUMZEETANCNFSM5IJ2CPUQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
-- 
Ahmad Anshorimuslim Syuhada
Data Engineer - Web Developer - Linux Enthusiast
+6285224752182 - @ans4175
"trust me I'm [already] an engineer"

imballinst commented:
> Kita ngunci di paketnya gitu ya terus bikin diff
> Boleh deh cobain
> Aku ga terlalu kebayang ji

iyak, betul :100: 

> Anyway, iyah tanpa border lebih masuk pas kayaknya
> Itu pakai wired card buatnya? Kok aneh ya

kayaknya si tabs ini di-desain untuk berada di "tempat kosong" (bukan di atas elemen wired)... soalnya bawaan dari `WiredTab`nya demikian, seperti di showcasenya

![image](https://user-images.githubusercontent.com/7077157/142630502-e3436b8c-1990-48d5-a951-1e965ee26782.png)

imballinst commented on patches/wired-elements+3.0.0-rc.6.patch:
```diff
@@ -0,0 +1,113 @@
++            this.fireSelected();
+         }
+     }
++    fireSelected() {
++        fireEvent(this, 'selected', { selected: this.selected });
++    }
```
@ans-4175 begini kurang lebih cara saya nambahin event emitter (?) di komponennya

ans-4175 replied:
okay, berarti nanti dibacanya di atasnya `onselected` sama seperti yang radio group button

imballinst replied:
iyak, betul

ans-4175 commented:
<img width="619" alt="image" src="https://user-images.githubusercontent.com/3961872/142708780-039b69fa-54ee-4d40-b6b9-57545b78674e.png">
Pas di randomize text, kalau ngerandom lagi dia lebih dahulu nampilin hasil dibanding randomnya beres.
Ada logic yang kelewat?

ans-4175 commented:
<img width="317" alt="image" src="https://user-images.githubusercontent.com/3961872/142708980-d62043f8-4d8a-4f98-a810-82403eb4199d.png">
Dilihat-lihat, apakah mungkin tab-textnya di tengah ya?

ans-4175 commented on package.json:
```diff
@@ -20,7 +20,8 @@
     "eject": "react-scripts eject",
-    "prettify": "prettier --write src"
+    "prettify": "prettier --write src",
+    "postinstall": "patch-package"
```
hoo baru tahu begini, berarti ini auto jalan?

imballinst replied:
iyak, jadi urutannya:

`yarn` --> `node_modules` terinstall (raw) --> `patch-package` (apply diff dari patches yang ada)

ans-4175 commented on patches/wired-elements+3.0.0-rc.6.patch:
```diff
@@ -0,0 +1,113 @@
+diff --git a/node_modules/wired-elements/lib/wired-lib.js b/node_modules/wired-elements/lib/wired-lib.js
```
cara bikinnya ini gimana ji? BIkin file perubahannya, terus di diff sendiri dan hasil diffnya masukkin sini?

ans-4175 replied:
naming filenya juga?

imballinst replied:
o ini ada caranya, maap, habis ini saya tambahin di READMEnya

ans-4175 commented on patches/wired-elements+3.0.0-rc.6.patch:
```diff
@@ -0,0 +1,113 @@
+     }
+     draw(svg, s) {
+-        rectangle(svg, 2, 2, s[0] - 4, s[1] - 4, this.seed);
++        if (this.hasBorder) {
```
haha nice

ans-4175 commented on patches/wired-elements-react+0.1.5.patch:
```diff
@@ -0,0 +1,25 @@
+ import { WiredTabs as CE } from 'wired-elements/lib/wired-tabs.js';
+-export const WiredTabs = createComponent(React, 'wired-tabs', CE);
++export const WiredTabs = createComponent(React, 'wired-tabs', CE, {
++  onselected: 'selected'
```
nice

ans-4175 commented on src/App.css:
```diff
@@ -11,7 +11,9 @@ wired-card {
-  max-width: 400px;
+  /* Make this the same with the wheel's width. */
+  /* That way, the wheel will be properly centered. */
+  max-width: 445px;
```
wheel size ini dia emang gabisa diubah sih ya, ok

imballinst replied:
saya belum riset mendalam, tapi mungkin bisa sih override `max-width` style di class `bhdLno`

ans-4175 commented on src/App.js:
```diff
@@ -77,16 +88,50 @@ function App() {
+      0,
+      optionsList[prizeNumber]['option'].length - 3
+    );
+    const pickedMerchant = merchants.find((merch) =>
```
nampaknya ada logic disini yang perlu dicek
<img width="527" alt="image" src="https://user-images.githubusercontent.com/3961872/142709237-c13cb439-506f-4857-8622-e9b33168ba95.png">
Dia jadi nampilin hasilnya dulu, sebelum spin-nya beres

imballinst replied:
ah iya dong, wut... my bad, saya cek habis ini

ans-4175 commented on src/App.js:
```diff
@@ -77,16 +88,50 @@ function App() {
+  };
+
+  const onChangeTab = (newTab) => {
+    setPickedMerchant(undefined);
```
ini kayaknya bikin satu higher order function lagi aja gimana?
Jadi dia sama, tapi nambah extend next function yang berbeda. Dibanding masing-masing punya states begini. Nanti sama aja kan behaviour states-nya

imballinst replied:
I see, iya sih, ada banyak setter calls yang duplicate di handler-handlernya. coba saya extract dan pilah-pilih. Thanks masukannya!

ans-4175 commented on src/App.js:
```diff
@@ -107,28 +150,13 @@ function App() {
-    setFetched(false);
-    setMustSpin(false);
-    refetch();
+    resetStates();
```
yap nice

ans-4175 commented on src/App.js:
```diff
@@ -107,28 +150,13 @@ function App() {
-    setFetched(true);
-    setPickedMenus(randomMenu);
-  };
+  const onFinishRandomizing = useCallback(() => {
```
kayaknya ini kecepetan deh kepanggilnya di component reactnya

imballinst replied:
bisa jadi, coba saya cek

ans-4175 commented on src/App.js:
```diff
@@ -169,84 +199,104 @@ function App() {
                   <Wheel
-                    mustStartSpinning={mustSpin}
+                    mustStartSpinning={
+                      randomizerMode === 'wheel' && mustStartRandomizing
```
oh nice

ans-4175 commented on src/App.js:
```diff
@@ -169,84 +199,104 @@ function App() {
-                  <p className="txt-resto text-center">{pickedMerchant.name}</p>
+              {detailMerchant !== undefined &&
+                pickedMerchant !== undefined &&
+                isMerchantDetailShown && (
```
coba cek logic `isMerchantDetailShown` ini keubah dimana ya? Jadi muncul duluan sebelum beres stop spin atau randomizer

ans-4175 commented on src/components/TextBlinkRandomizer.js:
```diff
@@ -0,0 +1,67 @@
+        <div>{optionsList[0].option}</div>
+
+        <div className="randomizer-text-helper">
+          Press "Randomize" button below to start.
```
wkwk, anti pattern. Dia berharap ada tombol click di bawah di luar dari component ini

imballinst replied:
wkwkwk, bener juga sih... apa kalau gitu mending dibuat gini?

1. Tombol spin dan tombol reset jadi 1 komponen sendiri: `SpinnerButtons`
2. Bikin komponen baru: `WheelRandomizer` -- yang ngeimport `SpinnerButtons`
3. Import `SpinnerButtons` di `TextBlinkRandomizer` yang ada sekarang
4. Delete spinner buttons yang ada di `App.js`

imballinst replied:
Saya jadinya pakai pattern `children`, kira-kira gimana? Saya senangnya pattern ini karena kita nggak perlu nambahin prop sih:

```
// App.js.
const spinnerButtons = mustStartRandomizing ? (
  <p>Waiting to {randomizeText}...</p>
) : (
  <div>
    <WiredButton elevation={2} onClick={handleSpinClick}>
      {randomizeText}
    </WiredButton>
    <WiredButton
      className="btn-reset"
      elevation={2}
      onClick={handleResetClick}
    >
      RE-LOAD
    </WiredButton>
  </div>
);

<TextBlinkRandomizer
  onFinishRandomizing={onFinishRandomizing}
  mustStartRandomizing={
    randomizerMode === 'text' && mustStartRandomizing
  }
  pickedMerchant={pickedMerchant}
  optionsList={optionsList}
>
  {spinnerButtons}
</TextBlinkRandomizer>

// TextBlinkRandomizer.js.
if (shownText === undefined) {
  return (
    <>
      <div className="randomizer-text-container">
        <div>{optionsList[0].option}</div>

        <div className="randomizer-text-helper">
          Press "Randomize" button below to start.
        </div>
      </div>

      {children}
    </>
  );
}
```

ans-4175 replied:
oh nice ini, dia berarti selalu dibungkus sesuatu. sip, betul jadi ga ngerubah prop sebelumnya

ans-4175 reviewed (changes requested):
Ada beberapa logic yang shown merchant detail ini, nampaknya callback react-nya ga pas cycle-nya

imballinst commented:
> <img alt="image" width="619" src="https://user-images.githubusercontent.com/3961872/142708780-039b69fa-54ee-4d40-b6b9-57545b78674e.png">
> Pas di randomize text, kalau ngerandom lagi dia lebih dahulu nampilin hasil dibanding randomnya beres. Ada logic yang kelewat?

...wah, my bad, OK saya cek lagi

> <img alt="image" width="317" src="https://user-images.githubusercontent.com/3961872/142708980-d62043f8-4d8a-4f98-a810-82403eb4199d.png">
> Dilihat-lihat, apakah mungkin tab-textnya di tengah ya?

o iya bener, karena semuanya tengah, kayaknya lebih tepat di tengah. saya cobain

ans-4175 commented:
Gabisa deng, dia classnya itu random generated eh

On Sat, 20 Nov 2021 at 09.16 Try Ajitiono ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/App.css
> <https://github.com/ans-4175/random-gofood/pull/14#discussion_r753620794>:
>
> > @@ -11,7 +11,9 @@ wired-card {
>  }
>
>  section {
> -  max-width: 400px;
> +  /* Make this the same with the wheel's width. */
> +  /* That way, the wheel will be properly centered. */
> +  max-width: 445px;
>
> saya belum riset mendalam, tapi mungkin bisa sih override max-width style
> di class bhdLno
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/ans-4175/random-gofood/pull/14#discussion_r753620794>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA6HIEB7QOU4RW4DX2KCF7LUM4AIRANCNFSM5IJ2CPUQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
-- 
Ahmad Anshorimuslim Syuhada
Data Engineer - Web Developer - Linux Enthusiast
+6285224752182 - @ans4175
"trust me I'm [already] an engineer"

imballinst commented:
oiya deng, dia pakai `styled`, hm. ini buildnya kenapa eror yak @ans-4175, boleh tolong dicekin?

ans-4175 commented on README.md:
````diff
@@ -17,6 +17,28 @@ Look for files in
+To apply the patches under the [patches](./patches) folder, do this in the root of this project:
+
+```shell
+yarn patch-package
````
`yarn postinstall` bukan harusnya?

imballinst replied:
o bener, `yarn postinstall` aja biar ngereference ke `scripts`, siap

ans-4175 commented on src/App.js:
```diff
@@ -77,27 +86,66 @@ function App() {
+    setDetailMerchant(undefined);
+    setMustStartRandomizing(true);
+    // Disable the tabs.
+    tabsRef.current.requestUpdate();
```
sip

ans-4175 commented on src/App.js:
```diff
@@ -138,10 +172,33 @@ function App() {
   }, [merchants]);
 
+  const randomizeText = randomizerMode === 'wheel' ? 'Spin' : 'Randomize';
+  const isMerchantDetailShown =
```
oh nice

ans-4175 commented on src/components/WheelRandomizer.js:
```diff
@@ -0,0 +1,25 @@
+import { Wheel } from 'react-custom-roulette';
+
+export default function WheelRandomizer({
```
nice pattern here

ans-4175 reviewed (approved):
but checking on build error first. Karena `patch-package`
<img width="950" alt="image" src="https://user-images.githubusercontent.com/3961872/142733334-ef9f4f04-a0a2-4176-b332-eddcc6a7ccc1.png">
Seharusnya urutannya?
yarn, yarn postinstall, gitu kan?

ans-4175 commented:
<img width="516" alt="image" src="https://user-images.githubusercontent.com/3961872/142733496-43af7dce-3061-4974-b6df-505d57daec29.png">
Ini logicnya kita cek lagi, aku `spin` dan `randomize`-nya belum sesuai show rendernya

ans-4175 commented:
<img width="752" alt="image" src="https://user-images.githubusercontent.com/3961872/142735537-f8a87139-7719-4378-8002-c1078865c1b4.png">
Sudah bisa, aku perbaiki command install di vercel-nya, biar bisa patch package

ans-4175 commented:
Aku agak bingung di patch, ini dia masalah pertama yang kutemui, `onselected` ini ga trigger event deh, aku coba pasang debug console.log function, ga kepanggil
<img width="336" alt="image" src="https://user-images.githubusercontent.com/3961872/142735747-ef606308-fddf-4708-8f61-0d979ab44fd4.png">
Aku belum terlalu ngerti bikin 'diff' itu

imballinst commented:
> Seharusnya urutannya?
> yarn, yarn postinstall, gitu kan?

iyak

> <img alt="image" width="516" src="https://user-images.githubusercontent.com/3961872/142733496-43af7dce-3061-4974-b6df-505d57daec29.png">
> Ini logicnya kita cek lagi, aku `spin` dan `randomize`-nya belum sesuai show rendernya

...bener, wah maaf :facepalm: kemarin bugnya adalah ketika beres spin, terjadi race condition sehingga detail merchantnya nggak muncul (karena `onChangeTab` entah kenapa kepanggil juga)... terus saya remove beberapa `fireSelected`, terlihat bisa, eh ternyata yang textnya ngaco

> <img alt="image" width="752" src="https://user-images.githubusercontent.com/3961872/142735537-f8a87139-7719-4378-8002-c1078865c1b4.png">
> Sudah bisa, aku perbaiki command install di vercel-nya, biar bisa patch package

siap, thanks!

imballinst commented:
@ans-4175  _harusnya_ sih udah bener, ya... tapi buildnya masih gagal, kenapa ya?

https://user-images.githubusercontent.com/7077157/142746269-66e82be1-6a9d-44ed-83ee-d66612520994.mov

saya pengen nambahin test, tapi ditolak sama `jest`, soalnya si `wired-elements` ini dia `target`nya `es2017` bukan `es2015` (atau lebih rendah)... jadi hasilnya masih ada keyword `export` (yang nggak ada di JS biasa) :sweat_smile:  https://github.com/rough-stuff/wired-elements/blob/master/tsconfig.json#L3
